oat-sa/extension-tao-itemqti
# Feat/aut 2013/graphic gap match UI maxscore feedback

**Related to:** [AUT-2013](https://oat-sa.atlassian.net/browse/AUT-2013)

**Description:**

Show warning message to Author on graphicGapMatch Interaction when the matchMax is set to 0 (infinite)

**Changes:**

- Change the matchMax default value for hotspots to 1.
- Change the matchMax default value for gapImg to 1.
- Show warning messages on the cases set on the ticket and for possible pairs.

**How to check:**

- Create an item and add a graphicGapMatch Interaction.
- Add an image for the interaction and images for the gapImg.
- Add some hotspots.
- When the response processing is set to “map response“.
AND set the match max of a hotspot [A] to 0 (infinite)
AND set match max of a gap image [B] to 0 (infinite)
AND set the score of the response map for the pair hotspot [A] / gap image [B] to a value > 0
THEN the warning message will be shown

**For case scenarios of possible pairs that are not already set:** 
- If the Mapping Default on Map response is set to 1, it will check the possible pairs with the pair scoring set to 1, so if the gap Image and hotspot have a value of 0, the warning will be shown as well. 

![image](https://user-images.githubusercontent.com/11692751/166684845-78fc7088-6597-49b1-ba8b-eade2ebbab2e.png)
